### PR TITLE
meson.eclass: EAPI 8 support

### PR DESCRIPTION
**Pull request by dm0-**
@floppym Here is an update to allow EAPI 8 with meson packages (excluding the `meson-multilib.eclass`, since that can be done with all of its multilib eclass dependencies).

This updates `meson.eclass` to conform to conventions that other eclasses seem to follow.  E.g. conditional inherits are first (presumably for function precedence), and defining the inherit guard at the end.  It also removes the split inherit guard.  The only reason I saw for it is [bug #619178 comment #4](https://bugs.gentoo.org/619178#c4) with no further details.  Are there actually any examples of this being a problem?  A couple Python eclasses seem to be the only other instances of this.

### CODE REVIEW

File: eclass/meson.eclass [hunk 1]
```diff
@@ -35,31 +35,24 @@
 # @CODE
 
 case ${EAPI:-0} in
```
Comment (ulm) [inline eclass/meson.eclass:37]:
```suggestion
case ${EAPI} in
```

**Review by ulm — APPROVED** [review]

File: eclass/meson.eclass (hunk 1 above)
Comment (dm0-) [reply eclass/meson.eclass]: Updated.  (This is another case where it seems most other eclasses are doing that.)

**Comment by floppym** [comment]
I would prefer to keep EXPORT_FUNCTIONS outside the include guard. It doesn't make the eclass that much more complex, and the resulting behavior is nicer.

**Comment by dm0-** [comment]
I've added a quick update to the inherited `eapi7-ver.eclass` so that it doesn't say EAPI 8 is unknown.

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**